activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -90,16 +90,21 @@ def expand_from_hash(attributes)
             queries.reduce(&:or)
           elsif table.aggregated_with?(key)
             mapping = table.reflect_on_aggregation(key).mapping
-            queries = Array.wrap(value).map do |object|
-              mapping.map do |field_attr, aggregate_attr|
-                if mapping.size == 1 && !object.respond_to?(aggregate_attr)
-                  build(table.arel_attribute(field_attr), object)
-                else
-                  build(table.arel_attribute(field_attr), object.send(aggregate_attr))
+            if mapping.length == 1
+              mapping.first.yield_self do |field_attr, aggregate_attr|
```

Comment:
Can we not use `yield_self`? This make harder to backport.

Reply:
Done.  Note that I needed to use different variable names from the `mapping.map` in the `else` branch to avoid "shadowing outer local variable" warnings.